queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,63 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
+
+
+;; Rule imports
+(rule_import
+  "use" @include
+  "rule" @include
+  "from" @include
+  "as"? @include
+  "with"? @include
+)
+
+
+;; Rule inheritance
+(rule_inheritance
+  "use" @keyword
+  "rule" @keyword
+  "with"? @keyword
+)
+
+
+;; Wildcard names
+;; as @field for consistency with attributes in wildcard interpolations
+(wildcard (identifier) @field)
+
+
+;; Keywords in wildcard interpolations
+(
+  (identifier) @keyword
+  (#has-ancestor? @keyword "directive")
+  (#has-ancestor? @keyword "block")
+  (#any-of? @keyword
+      "config"
+      "input"
+      "log"
+      "output"
+      "params"
+      "resources"
+      "threads"
+      "wildcards"
+   )
+)
+
+(
+  (identifier) @keyword
+  (#has-ancestor? @keyword "wildcard")
+  (#any-of? @keyword
+      "config"
+      "input"
+      "log"
+      "output"
+      "params"
+      "resources"
+      "threads"
+      "wildcards"
+   )
+)
```

Comment:
This seems unnecessarily costly, a code snippet w/ the corresponding ts tree would be nice to find better ways to add captures for it

Reply:
Here's an example. These keywords are context-specific and only have "special snakemake meaning" in particular contexts.
![tree](https://user-images.githubusercontent.com/8322329/237042739-77216cab-6348-4392-ba03-a10528f16f68.jpg)